Fix a syntax error when ((...)) is combined with redirections

JohnoKing opened the pull request:
The following command causes a spurious syntax error:
```
$ ((1+2)) > /dev/null
/usr/bin/ksh: syntax error: `>' unexpected
```
This happens because the `done` label in parse.c is placed after the `inout` call that handles I/O redirections; `>` is not treated as a redirection operator after `goto done`:
https://github.com/ksh93/ksh/blob/5e7d335f2fc9c42b26386c4eea7abe52da5944e5/src/cmd/ksh93/sh/parse.c#L1149-L1151
https://github.com/ksh93/ksh/blob/5e7d335f2fc9c42b26386c4eea7abe52da5944e5/src/cmd/ksh93/sh/parse.c#L1370-L1379

Moving the `done` label fixes the syntax error because `inout` needs to handle `>`. This bugfix was backported from ksh93v- 2013-10-10-alpha.

JohnoKing committed:
Fix a syntax error when ((...)) is combined with redirections

src/cmd/ksh93/sh/parse: item():
- The done label is placed after the 'inout' call for handling I/O
  redirections. This causes the command below to produce a syntax
  error because the '>' is not handled as a redirection operator
  after 'goto done':
  $ ((1+2)) > /dev/null
  /usr/bin/ksh: syntax error: `>' unexpected
  Moving the done label fixes the syntax error as 'inout' is now
  called to handle the redirection operator.

src/cmd/ksh93/tests/arith.sh:
- Add a regression test for combining redirections with ((...)).

This bugfix was backported from ksh93v- 2013-10-10-alpha.

JohnoKing committed:
Fix a typo in the error message

McDutchie commented on src/cmd/ksh93/tests/arith.sh:
```diff
@@ -761,4 +761,10 @@ x=0x1.0000000000000000000000000000p+6
 
+# ======
+# Redirections with ((...)) should not cause a syntax error
+"$SHELL" 2>/dev/null -c '(($(echo 1+1 | tee /dev/fd/3))) >/dev/null 3>&1'
```
`/dev/fd/` is not portable. I also don't understand what that whole command substitution is for. A simple way to test for this syntax error would be:
```sh
(eval '((1)) >/dev/null') 2>/dev/null || err_exit 'redirections with ((...)) yield a syntax error'
```
Using `eval` avoids triggering a syntax error prematurely on a ksh with the bug.

McDutchie commented on src/cmd/ksh93/sh/parse.c:
```diff
@@ -1367,13 +1367,13 @@ static Shnode_t	*item(Lex_t *lexp,int flag)
 	}
 	sh_lex(lexp);
+done:
 	/* redirection(s) following a compound command */
```
Since this is now also used for redirections following a simple arithmetic command, the comment should be updated.

JohnoKing committed:
Use a simpler regression test and update a comment

src/cmd/ksh93/tests/arith.sh
- Replace the regression test from ksh93v- with a simpler one
  that uses 'eval' (from Martijn Dekker).

src/cmd/ksh93/sh/parse.c:
- Update the redirections comment to indicate it is also used
  for arithmetic expressions.
